pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -1505,7 +1506,7 @@ def _get_toolset(
         toolset = CombinedToolset(toolsets)
 
         def copy_dynamic_toolsets(toolset: AbstractToolset[AgentDepsT]) -> AbstractToolset[AgentDepsT]:
-            if isinstance(toolset, DynamicToolset):
+            if isinstance(toolset, DynamicToolset) or isinstance(toolset, SearchableToolset):
```

Comment:
TODO: a Pythonic way to add a marker (protocol?) for toolsets that can prefer to copy their instance state for each id so the instance checks do not grow here.

Reply:
Ah I see why initial attempts fail here, the root SearchableToolset is constructed already, so to cover this we need to have the nested SearchableToolset instances.